utils/python/CIME/XML/compilerblock.py

```diff
@@ -115,6 +117,35 @@ def _handle_references(self, elem, set_up, tear_down, depends):
         output = elem.text
         if output is None:
             output = ""
+
+        reference_re = re.compile(r'\${?(\w+)}?')
```

Comment:
Why can't we reuse the code from generic_xml that already handles this kind of stuff?